src/libsystemd-network/sd-radv.c

```diff
@@ -569,6 +580,40 @@ int sd_radv_set_preference(sd_radv *ra, unsigned preference) {
         return 0;
 }
 
+int sd_radv_set_home_agent_information(sd_radv *ra, int home_agent) {
+        assert_return(ra, -EINVAL);
+
+        if (ra->state != RADV_STATE_IDLE)
+                return -EBUSY;
+
+        SET_FLAG(ra->flags, ND_RA_FLAG_HOME_AGENT, home_agent);
+
+        return 0;
+}
+
+int sd_radv_set_home_agent_preference(sd_radv *ra, uint16_t preference) {
+        assert_return(ra, -EINVAL);
+
+        if (ra->state != RADV_STATE_IDLE)
+                return -EBUSY;
+
+        ra->home_agent.nd_opt_home_agent_info_preference = htobe16(preference);
+
+        return 0;
+}
+
+int sd_radv_set_home_agent_lifetime(sd_radv *ra, uint16_t lifetime) {
+        assert_return(ra, -EINVAL);
+        assert_return(lifetime < RADV_MAX_HOME_AGENT_LIFETIME_USEC / USEC_PER_SEC, -EINVAL);
```

Comment:
This should be simple check, rather than `assert_return()`, even if the conf parser already checks the range:
```c
if (lifetime * USEC_PER_SEC >= RADV_MAX_HOME_AGENT_LIFETIME_USEC)
        return -EINVAL;
```